### PR TITLE
Support POST, body and custom headers for `fetch`

### PR DESCRIPTION
**Pull request by nieopierzony**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

File: lib/network.js [hunk 1]
```diff
@@ -3,12 +3,30 @@
 const http = require('http');
 const https = require('https');
 
-const fetch = async (url) => {
+const makeRequestOptions = (url, options = {}) => {
+  const { hostname, port, pathname, search } = new URL(url);
+  const path = `${pathname}${search}`;
+  const method = options.method ?? 'GET';
+  const headers = options.headers ?? {};
```
Comment (tshemsedinov) [inline lib/network.js:10]:
```suggestion
  const { method = 'GET', headers = {} } = options;
```

File: lib/network.js [hunk 2]
```diff
@@ -3,12 +3,30 @@
...
+  const result = { hostname, port, path, method, headers };
+  if (options.body) {
+    result.body = JSON.stringify(options.body);
+    result.headers = {
+      'Content-Type': 'application/json',
+      'Content-Length': Buffer.byteLength(result.body),
+      ...result.headers,
```
Comment (tshemsedinov) [inline lib/network.js:17]:
```suggestion
      ...headers,
```

File: lib/network.js (hunk 1 above)
Comment (tshemsedinov) [inline lib/network.js:10]:
```suggestion
const { method = 'GET', headers = {} } = options;
```

File: lib/network.js [hunk 3]
```diff
@@ -31,4 +51,4 @@ const receiveBody = async (req) => {
   return Buffer.concat(buffers);
 };
 
-module.exports = { fetch, receiveBody };
+module.exports = { fetch, receiveBody, makeRequestOptions };
```
Comment (tshemsedinov) [inline lib/network.js:54]: We do not need to export it separately
```suggestion
module.exports = { fetch, receiveBody };
```

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

File: lib/network.js (hunk 3 above)
Comment (nieopierzony) [reply lib/network.js:54]: That's what I thought too, but then its not possible to test it

**Comment by tshemsedinov** [comment]
Need to check compatibility with https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch

File: test/network.js [hunk 4]
```diff
@@ -17,9 +22,53 @@ const getRate = async (currency) => {
...
+    body: '{"a":1}',
+    headers: {
+      Test: 'test',
+      'Content-Type': 'application/json',
+      'Content-Length': 7,
+    },
+  });
+  test.end();
+});
+
```
Comment (tshemsedinov) [inline test/network.js:60]:
```suggestion

```

**Comment by tshemsedinov** [comment]
:partying_face: Thanks for contribution!

**Pull request merged**